contracts/zero-ex/contracts/src/features/UniswapFeature.sol

```diff
@@ -0,0 +1,320 @@
+    uint256 constant private FF_SUSHISWAP_FACTORY = 0xFFC0AEe478e3658e2610c5F7A4A2E1777cE9e4f2Ac0000000000000000000000;
+    uint256 constant private UNISWAP_PAIR_INIT_CODE_HASH = 0x96e8ac4277198ff8b6f785478aa9a39f403cb768dd02cbee326c3e7da348845f;
+    uint256 constant private SUSHISWAP_PAIR_INIT_CODE_HASH = 0xe18a34eb0e04b04f7a0ac29a6e80748dca96319b42c54d679cb821dca90c6303;
+    uint256 constant private ADDRESS_MASK = 0x000000000000000000000000ffffffffffffffffffffffffffffffffffffffff;
+    uint256 constant private ETH_TOKEN_ADDRESS_32 = 0x000000000000000000000000eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee;
+
+    uint256 constant private ALLOWANCE_TARGET_EXECUTE_CALL_SELECTOR_32 = 0xbca8c7b500000000000000000000000000000000000000000000000000000000;
+    uint256 constant private UNISWAP_PAIR_RESERVES_CALL_SELECTOR_32 = 0x0902f1ac00000000000000000000000000000000000000000000000000000000;
+    uint256 constant private UNISWAP_PAIR_SWAP_CALL_SELECTOR_32 = 0x022c0d9f00000000000000000000000000000000000000000000000000000000;
+    uint256 constant private TRANSFER_FROM_CALL_SELECTOR_32 = 0x23b872dd00000000000000000000000000000000000000000000000000000000;
+    uint256 constant private WETH_WITHDRAW_CALL_SELECTOR_32 = 0x2e1a7d4d00000000000000000000000000000000000000000000000000000000;
+    uint256 constant private WETH_DEPOSIT_CALL_SELECTOR_32 = 0xd0e30db000000000000000000000000000000000000000000000000000000000;
+    uint256 constant private ERC20_TRANSFER_CALL_SELECTOR_32 = 0xa9059cbb00000000000000000000000000000000000000000000000000000000;
+
+    /// @dev Construct this contract.
+    /// @param weth The WETH contract.
+    /// @param allowanceTarget The AllowanceTarget contract.
+    constructor(IEtherTokenV06 weth, IAllowanceTarget allowanceTarget) public {
+        WETH = weth;
+        ALLOWANCE_TARGET = allowanceTarget;
+    }
+
+    /// @dev Initialize and register this feature.
+    ///      Should be delegatecalled by `Migrate.migrate()`.
+    /// @return success `LibMigrate.SUCCESS` on success.
+    function migrate()
+        external
+        returns (bytes4 success)
+    {
+        _registerFeatureFunction(this.sellToUniswap.selector);
+        return LibMigrate.MIGRATE_SUCCESS;
+    }
+
+    /// @dev Efficiently sell directly to uniswap/sushiswap.
+    /// @param tokens Sell path.
+    /// @param sellAmount of `tokens[0]` Amount to sell.
+    /// @param minBuyAmount Minimum amount of `tokens[-1]` to buy.
+    /// @param isSushi Use sushiswap if true.
+    /// @return buyAmount Amount of `tokens[-1]` bought.
+    function sellToUniswap(
+        IERC20TokenV06[] calldata tokens,
+        uint256 sellAmount,
+        uint256 minBuyAmount,
+        bool isSushi
+    )
+        external
+        payable
+        override
+        returns (uint256 buyAmount)
+    {
+        require(tokens.length > 1, "UniswapFeature/InvalidTokensLength");
```

Comment:
Do we need to check that amounts don't exceed 2**112 as in https://github.com/0xProject/0x-monorepo/blob/feat/efficient-uniswap/contracts/zero-ex/contracts/src/features/DirectUniswap.sol#L43?

Reply:
Ah yeah, nice catch.